CMakeLists.txt

```diff
@@ -436,10 +436,18 @@ endif()
 set(CMAKE_REQUIRED_DEFINITIONS)
 
 set(CMAKE_REQUIRED_DEFINITIONS -D_ISOC11_SOURCE=1)
+set(CMAKE_REQUIRED_FLAGS)
+if(APPLE)
+    check_c_compiler_flag(-Wunguarded-availability-new HAVE_W_UNGUARDED_AVAILABILITY_NEW)
+    if(HAVE_W_UNGUARDED_AVAILABILITY_NEW)
+        set(CMAKE_REQUIRED_FLAGS "-Wunguarded-availability-new -Werror")
```

Comment:
It might be worth doing this when checking for posix_memalign as well.

Reply:
I think it makes sense to wrap all checks with these flags.

@Dead2, @nmoinvaz what do you think?

Reply:
> What do you mean exactly?

For all feature checks add:

```
set(CMAKE_REQUIRED_FLAGS "-Wunguarded-availability-new -Werror")
```

Without this flags on macOS cmake may falsely detect features that are not available in the target macOS version as available.

See: https://github.com/zlib-ng/zlib-ng/issues/1877#issuecomment-2676808178

Reply:
It seems like it might be a good idea to me.

Reply:
Implemented